<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: LLSimpute algorithm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for llsImpute {pcaMethods}"><tr><td>llsImpute {pcaMethods}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>LLSimpute algorithm</h2>

<h3>Description</h3>

<p>Missing value estimation using local least squares (LLS).  First,
k variables (for Microarrya data usually the genes)  are selected
by pearson, spearman or kendall correlation coefficients.  Then
missing values are imputed by a linear combination of the k
selected variables. The optimal combination is found by LLS
regression.  The method was first described by Kim et al,
Bioinformatics, 21(2),2005.
</p>


<h3>Usage</h3>

<pre>
llsImpute(Matrix, k = 10, center = FALSE, completeObs = TRUE,
  correlation = "pearson", allVariables = FALSE, maxSteps = 100,
  xval = NULL, verbose = FALSE, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>Matrix</code></td>
<td>
<p><code>matrix</code> &ndash; Data containing the variables
(genes) in columns and observations (samples) in rows. The data
may contain missing values, denoted as <code>NA</code>.</p>
</td></tr>
<tr valign="top"><td><code>k</code></td>
<td>
<p><code>numeric</code> &ndash; Cluster size, this is the number of
similar genes used for regression.</p>
</td></tr>
<tr valign="top"><td><code>center</code></td>
<td>
<p><code>boolean</code> &ndash; Mean center the data if TRUE</p>
</td></tr>
<tr valign="top"><td><code>completeObs</code></td>
<td>
<p><code>boolean</code> &ndash; Return the estimated complete
observations if  TRUE. This is the input data with NA values
replaced by the estimated values.</p>
</td></tr>
<tr valign="top"><td><code>correlation</code></td>
<td>
<p><code>character</code> &ndash; How to calculate the
distance between genes.  One out of pearson | kendall | spearman ,
see also help(&quot;cor&quot;).</p>
</td></tr>
<tr valign="top"><td><code>allVariables</code></td>
<td>
<p><code>boolean</code> &ndash; Use only complete genes to
do the regression if TRUE, all genes if FALSE.</p>
</td></tr>
<tr valign="top"><td><code>maxSteps</code></td>
<td>
<p><code>numeric</code> &ndash; Maximum number of iteration
steps if allGenes = TRUE.</p>
</td></tr>
<tr valign="top"><td><code>xval</code></td>
<td>
<p><code>numeric</code> Use LLSimpute for cross
validation. xval is the index of the gene to estimate, all other
incomplete genes will be ignored if this parameter is set. We do
not consider them in the cross-validation.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p><code>boolean</code> &ndash; Print step number and relative
change if TRUE and  allVariables = TRUE</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>Reserved for parameters used in future version of the
algorithm</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Missing values are denoted as <code>NA</code><br /> It is not recommended
to use this function directely but rather to use the nni() wrapper
function. The methods provides two ways for missing value
estimation, selected by the <code>allVariables</code> option. The first
one is to use only complete variables for the  regression. This is
preferable when the number of incomplete variables is relatively
small.
</p>
<p>The second way is to consider all variables as candidates for the
regression.  Hereby missing values are initially replaced by the
columns wise mean.  The method then iterates, using the current
estimate as input for the regression until the change between new
and old estimate falls below a threshold (0.001).
</p>


<h3>Value</h3>

<table summary="R valueblock">
<tr valign="top"><td><code>nniRes</code></td>
<td>
<p>Standard nni (nearest neighbour
imputation) result object of this package. See
<code><a href="nniRes.html">nniRes</a></code> for details.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Each step the generalized inverse of a <code>miss</code> x k
matrix is calculated. Where <code>miss</code> is the number of missing
values in  variable j and <code>k</code> the number of neighbours. This
may be slow for large values of k and / or many missing
values. See also help(&quot;ginv&quot;).
</p>


<h3>Author(s)</h3>

<p>Wolfram Stacklies
</p>


<h3>References</h3>

<p>Kim, H. and Golub, G.H. and Park, H.  - Missing value
estimation for DNA microarray gene expression data: local least
squares imputation.  <em>Bioinformatics, 2005; 21(2):187-198.</em>
</p>
<p>Troyanskaya O. and Cantor M. and Sherlock G. and Brown P. and
Hastie T. and Tibshirani R. and Botstein D. and Altman RB.  -
Missing value estimation methods for DNA microarrays.
<em>Bioinformatics. 2001 Jun;17(6):520-525.</em>
</p>


<h3>See Also</h3>

<p><code><a href="pca.html">pca</a>, <a href="nniRes.html">nniRes</a>, <a href="nni.html">nni</a></code>.
</p>


<h3>Examples</h3>

<pre>
## Load a sample metabolite dataset (metaboliteData) with already 5\% of
## data missing
data(metaboliteData)
## Perform llsImpute using k = 10
## Set allVariables TRUE because there are very few complete variables
result &lt;- llsImpute(metaboliteData, k = 10, correlation="pearson", allVariables=TRUE)
## Get the estimated complete observations
cObs &lt;- completeObs(result)
</pre>

<hr /><div style="text-align: center;">[Package <em>pcaMethods</em> version 1.78.0 <a href="00Index.html">Index</a>]</div>
</body></html>
